Sources/MapboxNavigation/BannerPresentation.swift

```diff
@@ -0,0 +1,237 @@
+import UIKit
+
+protocol BannerPresentation: UIViewController {
+    
+    var navigationView: NavigationView { get }
+    
+    var bannerPresentationDelegate: BannerPresentationDelegate? { get set }
+    
+    var topBanners: Stack<Banner> { get set }
+    
+    var bottomBanners: Stack<Banner> { get set }
+    
+    func topBanner(at position: BannerPosition) -> Banner?
+    
+    func popBanner(at position: BannerPosition,
+                   animated: Bool,
+                   duration: TimeInterval,
+                   animations: (() -> Void)?,
+                   completion: (() -> Void)?) -> Banner?
+    
+    func push(_ banner: Banner,
+              animated: Bool,
+              duration: TimeInterval,
+              animations: (() -> Void)?,
+              completion: (() -> Void)?)
+}
+
+extension BannerPresentation {
```

Comment:
I plan to use default implementation of `BannerPresentation` protocol in both `PreviewViewController` and `NavigationViewController`.